### PR TITLE
feat(cli): Validate plugins gRPC protocol version

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/3628

Examples:
image

image

### CODE REVIEW

File: cli/cmd/sync.go [hunk 1]
```diff
@@ -257,3 +261,31 @@ func syncConnectionV1(ctx context.Context, cqDir string, sourceClient *clients.S
...
+		destPluginVersion, err := destClient.Version(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get destination version: %w", err)
+		}
+
+		destProtocolVersion, err := destClient.GetProtocolVersion(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get protocol version from destination %q: %w", destPluginName+"@"+destPluginVersion, err)
+		}
+		if destProtocolVersion != expectedProtocolVersion {
```
Comment (erezrokah) [inline cli/cmd/sync.go:282]: Initially I thought about trying to be more specific (e.g. telling the user to update only the destination or only the source), but that probably has a ton of edge cases. For example what if we have protocol version 3, then we need to start to check ranges 🙃

**Review by disq — APPROVED** [review]

**Review by yevgenypats — CHANGES_REQUESTED** [review]
I think let's wait with that. it should be done on the server side where there are still few culprits before we can take care of that.

Mostly we need to start versioning SDK with v1, v2 and making protobuf versioned as well (the package name in prtobouf) - this way v1 will never be able to connect to an uncompatible version.

**Comment by erezrokah** [comment]
> I think let's wait with that. it should be done on the server side where there are still few culprits before we can take care of that.
> 
> Mostly we need to start versioning SDK with v1, v2 and making protobuf versioned as well (the package name in prtobouf) - this way v1 will never be able to connect to an uncompatible version.

We can always remove this logic when we have server side validation. This is mostly needed for users updating to the latest CLI version with the type system changes (we've seen reports on Discord with the error mentioned in the bug)

**Comment by erezrokah** [comment]
Also the tests are blocked by https://github.com/cloudquery/plugin-sdk/issues/351 but we can comment them

**Comment by erezrokah** [comment]
PR to fix the Windows bug https://github.com/cloudquery/plugin-sdk/pull/352

**Comment by yevgenypats** [comment]
Im just working on this atm in my sdk pr https://github.com/cloudquery/plugin-sdk/pull/348 so this will create more work for me reviewing and then removing the logic. let's hold off

**Comment by yevgenypats** [comment]
Let's close this?

**Pull request closed without merging**